app/uk/gov/hmrc/helptosavestub/controllers/EligibilityCheckController.scala

```diff
@@ -42,32 +43,55 @@ class EligibilityCheckController extends BaseController {
-      }
-
   val alreadyHasAccountResult: EligibilityCheckResult =
     EligibilityCheckResult("HtS account already exists", 3, "HtS account was previously created", 1)
 
+  val invalidResultCode: EligibilityCheckResult =
+    EligibilityCheckResult("INVALID RESULT WHICH DES SHOULD NEVER SEND", 4, "Not entitled to WTC and not in receipt of UC", 2)
+
+  def eligibleResult(reasonCode: Int): EligibilityCheckResult = {
+    val reason = reasonMappings.getOrElse(reasonCode, sys.error(s"Could not find eligibility reason for code $reasonCode"))
+    EligibilityCheckResult("Eligible to HtS Account", 1, reason, reasonCode)
+  }
+
+  def ineligibleResult(reasonCode: Int): EligibilityCheckResult = {
+    val reason = reasonMappings.getOrElse(reasonCode, sys.error(s"Could not find eligibility reason for code $reasonCode"))
+    EligibilityCheckResult("Ineligible to HtS Account", 2, reason, reasonCode)
+  }
+
+  def getReasonCodeFromNino(nino: String): Int =
+    Try(nino.substring(3, 4).toInt)
+      .getOrElse(sys.error(s"Error getting reason code from fourth character of NINO $nino"))
+
   def eligibilityCheck(nino: String): Action[AnyContent] = Action { implicit request ⇒
     val response: Option[EligibilityCheckResult] =
-      if (nino.startsWith("AC")) {
-        // if nino start with AC return someone who has already opened an account in the past
+      // Comments are for the Test & Release Services (T&RS) team
+      // Private BETA:
+      // Start NINO with EL07 to specify an eligible applicant in receipt of WTC (with reason code 7)
+      //
+      // After private BETA:
+      // Start NINO with EL06 to specify an eligible applicant in receipt of UC (with reason code 6)
+      // Start NINO with EL08 to specify an eligible applicant in receipt of WTC and UC (with reason code 8)
+      if (nino.toUpperCase().startsWith("EL")) {
+        Some(eligibleResult(getReasonCodeFromNino(nino)))
+      } // Private BETA:
+      // Start NINO with NE02 to specify an eligible applicant in receipt of WTC (with reason code 2)
+      // Start NINO with NE03 to specify an eligible customer in receipt of WTC (with reason code 3)
+      //
+      // After private BETA:
+      // Start NINO with NE06 to specify an eligible applicant in receipt of UC (with reason code 6)
+      // Start NINO with NE08 to specify an eligible applicant in receipt of WTC and UC (with reason code 8)
+      else if (nino.toUpperCase().startsWith("NE")) {
+        Some(ineligibleResult(getReasonCodeFromNino(nino)))
+      } // Start NINO with AC to specify an existing account holder (with reason code 1)
+      else if (nino.startsWith("AC")) {
         Some(alreadyHasAccountResult)
-      } else if (nino.toUpperCase().startsWith("NA")) {
-        ineligibleReasonGen.seeded(nino)
-      } else {
-        eligibleReasonGen.seeded(nino)
+      } // Start NINO with EE to specify an invalid result code
+      else if (nino.startsWith("EE")) {
```

Comment:
why do need this ?  this means that for specific ninos DES always returns invalid result codes ?

Reply:
Yes - I'd like the ability to check that we handle incorrect values from DES gracefully.